clumper/clump.py

```diff
@@ -62,6 +63,63 @@ def read_json(cls, path):
             return Clumper(data)
         return Clumper(json.loads(pathlib.Path(path).read_text()))
 
+    @classmethod
+    def read_csv(cls, path, delimiter=",", fieldnames=None, nrows=None):
+        """
+        Reads in a csv file. Can also read files from url.
+
+        Arguments
+        ---------
+        path : filename or url
+        delimiter: must be a single character. `,` is the default.
+        fieldnames: You may prefer a different set of keys for the data, in which case, you can supply new keys with the fieldnames.
+        By default, the first row of the csv will provide the Clumper keys if fieldnames is None. If fieldnames is provided,
+        then the first row stays as part of the data. You should ensure that the correct number of fieldnames is supplied,
+        as an incorrect number can lead to truncation of the clumper. So, if you have seven columns and your fieldnames length is 3,
+        then every row will have only 3 values, the remaining four will be cut off.
+        nrows: Number of rows to read in. Useful when reading large files. If `None`, all rows are read.
+
+        Usage:
+
+        ```python
+        from clumper import Clumper
+
+        clump = Clumper.read_csv("tests/monopoly.csv")
+        assert len(clump) == 22
+
+        clump = Clumper.read_csv("tests/monopoly.csv", nrows = 10)
+        assert len(clump) == 10
+
+        clump = Clumper.read_csv("https://calmcode.io/datasets/monopoly.csv")
+        assert len(clump) == 22
+
+        # By default, the first row of the csv is treated as the keys of the Clumper.
+        # If the fieldnames argument is not None, then the first row stays as part of the data.
+        fieldnames = ['date', 'currency', 'country', 'price', 'dollar_rate', 'cost']
+        clump = Clumper.read_csv("https://calmcode.io/datasets/bigmac.csv",
```

Comment:
Is there a reason you're using the bigmac dataset here and now the monopoly one?

Reply:
Obvious reason is I'm lazy to type in all the headers.

Reply:
Ah monopoly has way more headers? I suppose it's fine then.

Reply:
Shorter examples are better in the docstrings.